### DISCUSSION TITLE
Is there a way to grab a `SharedGlobals` instance in C++?

### DISCUSSION THREAD

**Discussion by D4yvid** [slint-ui/slint#6366] [Q&A] [answered]
I'm using Slint with C++, and i created a global object inside Slint, when i see the generated header, the `SharedGlobals` class has it in it, but i do not see any way of obtaining that value, is that possible?

**Comment by ogoffart** [comment]
You can access each global separately with the global() function: https://releases.slint.dev/1.8.0/docs/cpp/generated_code.html#global-singletons
Is there a particular reason you need access to the SharedGlobals?

**Reply by D4yvid** [reply]
Ohhh! This method is not being generated if i export this type inside another .slint file without a window attached to it:
```cmake
slint_target_sources(mui3-test.slint
	ui/material-theme.slint
	app.slint)
```
`ui/material-theme.slint`:
```slint
// ...
export { ColorPallete, ColorScheme } from "./base/ColorPallete.slint";
export { Typography } from "./base/Typography.slint";
export { Animator } from "./base/Animator.slint";
// ...
```

It only generates this method if i explicitly export this global in the `app.slint` file, that contains the app window:
```slint
import {
    MaterialSurface,
    MaterialText,
    MaterialWindow,
    ElevatedButton,
    ColorPallete,
    ColorScheme,
    ColorPallete,
    Typography,
    FilledButton,
    FilledTonalButton,
    OutlinedButton,
    TextButton
} from "ui/material-theme.slint";
import { HorizontalBox } from "std-widgets.slint";

export { ColorPallete }
```

Is that the desired behaviour? Needing to export explicitly on the slint file where it has a window instance?

**Reply by ogoffart** [reply, marked as answer]
Right, you need to re-export the globals from the main file.
Globals that are not exported are condiered as internal and their property might be optimized away.

**Reply by D4yvid** [reply]
Ohhh, now i understand! Thanks for the response (:
